build/build.sh

```diff
@@ -0,0 +1,187 @@
+    echo ""
+    echo "or"
+    echo ""
+    echo "Params usage: $0 [options...]" >&2
+    echo
+    echo "   -a, --deepdetect-arch          Choose Deepdetect architecture : ${deepdetect_arch[*]}"
+    echo "   -b, --deepdetect-build         Choose Deepdetect build profile : CPU (${deepdetect_cpu_build_profiles[*]}) / GPU (${deepdetect_gpu_build_profiles[*]})"
+    echo
+    exit 1
+}
+
+# Parse arguments
+while (("$#")); do
+    case "$1" in
+    -a | --deepdetect-arch)
+        DEEPDETECT_ARCH=$2
+        shift 2
+        ;;
+    -b | --deepdetect-build)
+        DEEPDETECT_BUILD=$2
+        shift 2
+        ;;
+    -h | --help)
+        help_menu
+        break
+        ;;
+    esac
+done
+
+# Deepdetect platform selector
+select_platform() {
+    clear
+    echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+    echo " Select Deepdetect build platform"
+    echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+    for index in ${!deepdetect_arch[*]}; do
+        echo "  $index. ${deepdetect_arch[$index]^^}"
+    done
+    echo ""
+
+    local choice
+    read -p "Enter choice : " choice
+    DEEPDETECT_ARCH=${deepdetect_arch[$choice]}
+}
+
+# Deepdetect select CPU build profile
+select_cpu_build_profile() {
+    clear
+    echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+    echo " Select Deepdetect build profile"
+    echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+    for index in ${!deepdetect_cpu_build_profiles[*]}; do
+        echo "  $index. ${deepdetect_cpu_build_profiles[$index]}"
+    done
+    echo ""
+
+    local choice
+    read -p "Enter choice : " choice
+    DEEPDETECT_BUILD=${deepdetect_cpu_build_profiles[$choice]}
+}
+
+# Deepdetect select GPU build profile
+select_gpu_build_profile() {
+    clear
+    echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+    echo " Select Deepdetect build profile"
+    echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+    for index in ${!deepdetect_gpu_build_profiles[*]}; do
+        echo "  $index. ${deepdetect_gpu_build_profiles[$index]}"
+    done
+    echo ""
+    
+    local choice
+    read -p "Enter choice : " choice
+    DEEPDETECT_BUILD=${deepdetect_gpu_build_profiles[$choice]}
+}
+
+# Use menu...
+show_interactive_platform_selector() {
+    # Main menu handler loop
+    select_platform
+    if [ ${DEEPDETECT_ARCH} == "cpu" ]; then
+        select_cpu_build_profile
+    fi
+    if [ ${DEEPDETECT_ARCH} == "gpu" ]; then
+        select_gpu_build_profile
+    fi
+}
+
+# Build functions
+cpu_build() {
+
+    case ${DEEPDETECT_BUILD} in
+
+    "caffe-tf")
+        cmake .. -DUSE_TF=ON -DUSE_TF_CPU_ONLY=ON -DUSE_SIMSEARCH=ON -DUSE_TSNE=ON -DUSE_NCNN=OFF
+        make -j
+        ;;
+
+    "armv7")
+        cmake .. -DUSE_NCNN=ON -DRPI3=ON -DUSE_HDF5=OFF -DUSE_CAFFE=OFF
+        make -j
+        ;;
+
+    *)
+        cmake .. -DUSE_XGBOOST=ON -DUSE_SIMSEARCH=ON -DUSE_TSNE=ON -DUSE_NCNN=ON
+        make -j
+        ;;
+    esac
+
+}
+
+gpu_build() {
+
+    case ${DEEPDETECT_BUILD} in
+
+    "caffe-cpu-tf")
```

Comment:
@quadeare this build case must be moved to `cpu_build()`

Reply:
Arf... My bad, that's fixed : https://github.com/jolibrain/deepdetect/pull/706/commits/08e8a1fbd1f3b0c525b3e203b861487f750e25e3